nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/manager/impl/WebClusterManagerCoordinator.java

```diff
@@ -0,0 +1,164 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.cluster.manager.impl;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.nifi.cluster.coordination.ClusterCoordinator;
+import org.apache.nifi.cluster.coordination.node.DisconnectionCode;
+import org.apache.nifi.cluster.coordination.node.NodeConnectionState;
+import org.apache.nifi.cluster.coordination.node.NodeConnectionStatus;
+import org.apache.nifi.cluster.node.Node;
+import org.apache.nifi.cluster.node.Node.Status;
+import org.apache.nifi.cluster.protocol.ConnectionRequest;
+import org.apache.nifi.cluster.protocol.NodeIdentifier;
+import org.apache.nifi.reporting.Severity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class WebClusterManagerCoordinator implements ClusterCoordinator {
+    private static final Logger logger = LoggerFactory.getLogger(WebClusterManagerCoordinator.class);
+
+    private final WebClusterManager clusterManager;
+
+    public WebClusterManagerCoordinator(final WebClusterManager clusterManager) {
+        this.clusterManager = clusterManager;
+    }
+
+    @Override
+    public void requestNodeConnect(final NodeIdentifier nodeId) {
+        final Node node = clusterManager.getRawNode(nodeId.getId());
+
+        if (node == null) {
+            final ConnectionRequest connectionRequest = new ConnectionRequest(nodeId);
+            clusterManager.requestConnection(connectionRequest);
+        } else {
+            node.setStatus(Status.DISCONNECTED);
+            clusterManager.requestReconnection(nodeId.getId(), "Anonymous");
+        }
+    }
+
+    @Override
+    public void finishNodeConnection(final NodeIdentifier nodeId) {
+        final Node node = clusterManager.getRawNode(nodeId.getId());
+        if (node == null) {
+            logger.error("Attempting to Finish Node Connection but could not find Node with Identifier {}", nodeId);
+            return;
+        }
+
+        node.setStatus(Status.CONNECTED);
+    }
+
+    @Override
+    public void requestNodeDisconnect(final NodeIdentifier nodeId, final DisconnectionCode disconnectionCode, final String explanation) {
+        try {
+            clusterManager.requestDisconnection(nodeId, false, explanation);
+
+            if (disconnectionCode == DisconnectionCode.LACK_OF_HEARTBEAT) {
+                final Node node = clusterManager.getRawNode(nodeId.getId());
+                if (node != null) {
+                    node.setHeartbeatDisconnection();
+                }
+            }
+        } catch (final Exception e) {
+            logger.error("Failed to request node {} disconnect from cluster due to {}", nodeId, e);
+            logger.error("", e);
```

Comment:
The first line logs a toString() of the Exception, whereas the second line logs the stack trace. Personally, I find the log messages easier to read this way when someone copies & pastes it, as they often do not look to grab a stack trace but if the toString() of the Exception is part of the original message, it makes it more clear.